rest-service/manager_rest/rest/resources_v3_1/agents.py

```diff
@@ -43,3 +48,77 @@ def get(self, pagination=None):
             node_ids=args.get('node_ids'),
             node_instance_ids=args.get('node_instance_ids'),
             install_method=args.get('install_methods'))
+
+
+class AgentsName(SecuredResource):
+    @rest_decorators.exceptions_handled
+    @rest_decorators.marshal_with(models.Agent)
+    @authorize('agent_get')
+    def get(self, name):
+        """
+        Get agent by name
+        """
+        validate_inputs({'name': name})
+        return get_storage_manager().get(models.Agent, name)
+
+    @rest_decorators.exceptions_handled
+    @rest_decorators.marshal_with(models.Agent)
+    @authorize('agent_create')
+    def put(self, name):
+        """
+        Create a new agent
+        """
+        request_dict = get_json_and_verify_params({
+            'node_instance_id': {'type': unicode},
+            'state': {'type': unicode}
+        })
+        validate_inputs({'name': name})
+        state = request_dict.get('state')
+        self._validate_state(state)
+        timestamp = utils.get_formatted_timestamp()
+        new_agent = models.Agent(
+            id=name,
+            name=name,
+            ip=request_dict.get('ip'),
```

Comment:
Why are we getting all of this data form the request, and not from the instance runtime properties?